web/src/pages/user.jsx

```diff
@@ -56,12 +85,18 @@ export default ({songs}) => {
 
     useEffect(() => {
         console.log("searched: ", searchedSongs);
-        if (!songs) setSearchedSongs(undefined);
+        if (!songs) {
+            setSearchedSongs(undefined);
+        }
         if (searchQuery === "") {
             setSearchedSongs(songs);
         } else {
             setSearchedSongs(fuse.search(searchQuery).map((result) => result.item));
         }
+
+        if (searchQuery !== "" && searchedSongs.length === 0 && !loading) {
```

Comment:
<img width="1354" alt="image" src="https://github.com/timrolsh/Songscope/assets/68395788/02e1002d-0fbe-46a5-a2ad-ab8db568e8c1">

There seems to be an issue with leaking search requests in this code... this results in poor css rendering and drastic performance impacts. I suspect it's something with this but I'm not 100% certain, would need more diagnosing. From my minimal testing, it appears to only happen when searches are happening.

Reply:
Fixed by adding Search Spotify button which allows users to control when requests are being made (no longer blended with local search)